### PR TITLE
Dịch "linear-regression-gluon_vn" - Phần 6 

### LINKED ISSUE #669: Dịch "linear-regression-gluon_vn" - Phần 6
Phần cần dịch: https://github.com/aivivn/d2l-vn/blame/master/chapter_linear-networks/linear-regression-gluon_vn.md#L256-L306

### PR DESCRIPTION
**Pull request by NhatQuanBui**
## Pull Request Checklist

**Đánh dấu tick (sau khi "Create Pull Request") vào các mục dưới đây:**

* [x] Pull Request này tương ứng với issue nào? Trả lời số issue sau dấu `#` (không có dấu cách): Close #669 

* [x] Bản dịch này có bám sát **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)** không? Nếu một từ/cụm chưa có trong Bảng thuật ngữ, hãy tạo một Pull Request riêng đề xuất cách dịch từ/cụm đó.

* [x] Format của bản gốc có được giữ nguyên không?
    * Các phần in nghiêng, in đậm.
    * Tiêu đề (số lượng dấu `#` đầu dòng).
    * Bảng biểu, chú thích cho bảng (dòng phía trên mỗi bảng bắt đầu bằng dấu `:`).
    * Dịch các từ trong hình vẽ nếu cần. Các dòng có hình có dạng: `caption`.
    * Dịch các chú thích hình vẽ (thay các cụm `*dịch chú thích ảnh phía trên*` bằng bản dịch tiếng Việt).
    * Không dịch các phần code (nằm giữa hai dấu `````).
    * Copy các công thức toán từ bản gốc (các đoạn có `$`).
    * Giữ các dòng gán nhãn (bắt đầu với `:label:`).
    * Không tự thêm bớt các dòng trắng.
     
* [x] Trong một đoạn văn, mỗi câu văn đã được viết trong một dòng, giữa các dòng không có dòng trắng. Markdown sẽ coi những dòng liền nhau không có dòng trắng là một đoạn văn. Việc này giúp công đoạn review được thuận tiện hơn.

* [x] Tất cả chữ trong các hình vẽ đã được dịch. Nếu chưa, vui lòng xem thêm **[hướng dẫn hoặc yêu cầu dịch hình](https://github.com/aivivn/d2l-vn/blob/master/img/README.md)**.

* [x] Điền tên của bạn vào mục "Những người thực hiện" ở cuối file.

* [x] **Sau khi được approve**, điền tên những reviewers mà bạn thấy có nhiều đóng góp cho Pull Request này. Xem danh sách Contributors **[tại đây](https://github.com/aivivn/d2l-vn/blob/master/docs/contributors_info.md)**.

## Với Reviewer

1. Chỉ review nếu Pull Request này đã sẵn sàng cho việc review.

2. **Bạn có thể đề xuất thay đổi cách dịch mỗi dòng trực tiếp như trong hình dưới đây:**
img

### CODE REVIEW

File: chapter_linear-networks/linear-regression-gluon_vn.md [hunk 1]
```diff
@@ -268,29 +268,34 @@ Once we start working with much more complex models, Gluon's advantages will gro
 However, once we have all the basic pieces in place, the training loop itself is strikingly similar to what we did when implementing everything from scratch.
 -->
 
-*dịch đoạn phía trên*
+Bạn có thể thấy rằng việc thể hiện mô hình của chúng ta thông qua Gluon đòi hỏi một vài dòng lệnh tương đối.
+Chúng ta cần phải phân bổ các tham số một cách riêng lẻ, xác định hàm số mất mát của chúng ta , hay hạ gradient ngẫu nhiên.
+Một khi chúng ta làm với nhiều mô hình rất phức tạp , những lợi ích của Gluon sẽ tăng lên nhiều.
```
Comment (honghanhh) [inline chapter_linear-networks/linear-regression-gluon_vn.md]:
```suggestion
Một khi chúng ta bắt đầu làm việc với các mô hình phức tạp hơn, những lợi ích của Gluon sẽ tăng lên đáng kể.
```

File: chapter_linear-networks/linear-regression-gluon_vn.md [hunk 2]
```diff
@@ -268,29 +268,34 @@ Once we start working with much more complex models, Gluon's advantages will gro
...
+Tuy nhiên, một khi chúng ta có tất cả các mảnh cơ bản ở đúng chỗ, vòng huấn luyện của chính bản thân rất giống với những gì chúng ta đã thực hiện tự lúc đầu.
```
Comment (honghanhh) [inline chapter_linear-networks/linear-regression-gluon_vn.md]:
```suggestion
Tuy nhiên, một khi chúng ta có tất cả các mảnh cơ bản đặt đúng chỗ, vòng huấn luyện sẽ tương tự như những gì ta đã làm khi triển khai mọi thứ từ đầu.
```

File: chapter_linear-networks/linear-regression-gluon_vn.md [hunk 3]
```diff
@@ -268,29 +268,34 @@ Once we start working with much more complex models, Gluon's advantages will gro
...
 
 <!--
 To refresh your memory: for some number of epochs, we will make a complete pass over the dataset (train_data), iteratively grabbing one minibatch of inputs and the corresponding ground-truth labels.
 For each minibatch, we go through the following ritual:
 -->
 
-*dịch đoạn phía trên*
+Để làm mới lại bộ nhớ của bạn: đối với một số epoch, chúng ta sẽ thực hiện qua tập dữ liệu (train_data), cứ lấy lặp đi lặp lại một minibatch của dữ liệu dầu vào và gán các nhãn gốc tương ứng.
```
Comment (honghanhh) [inline chapter_linear-networks/linear-regression-gluon_vn.md]:
```suggestion
Để làm mới lại bộ nhớ của bạn: đối với một số epoch, chúng ta sẽ thực hiện qua tập dữ liệu (train_data), cứ lấy lặp đi lặp lại một minibatch của dữ liệu đầu vào và gán các nhãn gốc tương ứng.
```

Comment (honghanhh) [reply chapter_linear-networks/linear-regression-gluon_vn.md]: "To refresh your memory" dịch "Để làm mới lại bộ nhớ của bạn:" nghe hơi word-to-word nhỉ, không biết có cách dịch nào hay hơn không ạ?

File: chapter_linear-networks/linear-regression-gluon_vn.md [hunk 4]
```diff
@@ -268,29 +268,34 @@ Once we start working with much more complex models, Gluon's advantages will gro
...
 
 <!--
 * Generate predictions by calling `net(X)` and calculate the loss `l` (the forward pass).
 * Calculate gradients by calling `l.backward()` (the backward pass).
 * Update the model parameters by invoking our SGD optimizer (note that `trainer` already knows which parameters to optimize over, so we just need to pass in the minibatch size.
 -->
 
-*dịch đoạn phía trên*
+Tạo những dự đoán bằng cách gọi `net(X)` và tính sự mất mát `l` (đường chuyền về phía trước)
+Tính các gradient bằng cách gọi `l.backward()` (đường chuyền về sau(
```
Comment (honghanhh) [inline chapter_linear-networks/linear-regression-gluon_vn.md]:
```suggestion
Tính các gradient bằng cách gọi `l.backward()` (đường chuyền về sau).
```

File: chapter_linear-networks/linear-regression-gluon_vn.md [hunk 5]
```diff
@@ -268,29 +268,34 @@ Once we start working with much more complex models, Gluon's advantages will gro
...
+Đối với mỗi minibatch, chúng ta cần tuân thủ theo trình tự:
 
 <!--
 * Generate predictions by calling `net(X)` and calculate the loss `l` (the forward pass).
 * Calculate gradients by calling `l.backward()` (the backward pass).
 * Update the model parameters by invoking our SGD optimizer (note that `trainer` already knows which parameters to optimize over, so we just need to pass in the minibatch size.
 -->
 
-*dịch đoạn phía trên*
+Tạo những dự đoán bằng cách gọi `net(X)` và tính sự mất mát `l` (đường chuyền về phía trước)
```
Comment (honghanhh) [inline chapter_linear-networks/linear-regression-gluon_vn.md]:
```suggestion
Tạo những dự đoán bằng cách gọi `net(X)` và tính sự mất mát `l` (đường chuyền về phía trước).
```

File: chapter_linear-networks/linear-regression-gluon_vn.md (hunk 3 above)
Comment (lyphilong) [reply chapter_linear-networks/linear-regression-gluon_vn.md]: @honghanhh mình nghĩ chỗ "To refresh your memory" nên dịch là "Để giúp bạn có thể nhớ thì"

File: chapter_linear-networks/linear-regression-gluon_vn.md (hunk 2 above)
Comment (lyphilong) [reply chapter_linear-networks/linear-regression-gluon_vn.md]: @honghanhh  @NhatQuanBui  "once we have all the basic pieces in place," mình nghĩ nên dịch thành "một khi ta nắm được rõ các điều cơ bản," nghe có vẻ ổn hơn ạ.

File: chapter_linear-networks/linear-regression-gluon_vn.md [hunk 6]
```diff
@@ -268,29 +268,34 @@ Once we start working with much more complex models, Gluon's advantages will gro
 However, once we have all the basic pieces in place, the training loop itself is strikingly similar to what we did when implementing everything from scratch.
 -->
 
-*dịch đoạn phía trên*
+Bạn có thể thấy rằng việc thể hiện mô hình của chúng ta thông qua Gluon đòi hỏi một vài dòng lệnh tương đối.
```
Comment (lkhphuc) [inline chapter_linear-networks/linear-regression-gluon_vn.md]:
```suggestion
Bạn có thể thấy rằng việc thể hiện mô hình của chúng ta thông qua Gluon đòi hỏi tương đối ít các dòng lệnh.
```

File: chapter_linear-networks/linear-regression-gluon_vn.md (hunk 2 above)
Comment (lkhphuc) [reply chapter_linear-networks/linear-regression-gluon_vn.md]:
```suggestion
Tuy nhiên, một khi ta nắm được rõ các điều cơ bản, vòng huấn luyện tự thân nó sẽ trông rất giống với những gì chúng ta tự thực hiện từ đầu.
```

File: chapter_linear-networks/linear-regression-gluon_vn.md (hunk 3 above)
Comment (lkhphuc) [reply chapter_linear-networks/linear-regression-gluon_vn.md]:
```suggestion
Để nhắc lại: cho một vài epoch, chúng ta sẽ duyêt qua toàn bộ tập dữ liệu (train_data) bằng cách lấy theo từng minibatch của dữ liệu đầu vào cộng với các nhãn gốc tương ứng.
```

File: chapter_linear-networks/linear-regression-gluon_vn.md (hunk 5 above)
Comment (lkhphuc) [reply chapter_linear-networks/linear-regression-gluon_vn.md]:
```suggestion
* Tạo những dự đoán bằng cách gọi `net(X)` và tính sự mất mát `l` (đường chuyền tới).
```
Bạn lưu ý giữ format nhé.

File: chapter_linear-networks/linear-regression-gluon_vn.md (hunk 4 above)
Comment (lkhphuc) [reply chapter_linear-networks/linear-regression-gluon_vn.md]:
```suggestion
* Tính các gradient bằng cách gọi `l.backward()` (đường chuyền ngược).
```

File: chapter_linear-networks/linear-regression-gluon_vn.md [hunk 7]
```diff
@@ -268,29 +268,34 @@ Once we start working with much more complex models, Gluon's advantages will gro
...
 <!--
 * Generate predictions by calling `net(X)` and calculate the loss `l` (the forward pass).
 * Calculate gradients by calling `l.backward()` (the backward pass).
 * Update the model parameters by invoking our SGD optimizer (note that `trainer` already knows which parameters to optimize over, so we just need to pass in the minibatch size.
 -->
 
-*dịch đoạn phía trên*
+Tạo những dự đoán bằng cách gọi `net(X)` và tính sự mất mát `l` (đường chuyền về phía trước)
+Tính các gradient bằng cách gọi `l.backward()` (đường chuyền về sau(
+Cập nhật các tham số của mô hình bằng cách dẫn các tối ưu hóa của hạ gradient ngẫu nhiên (SGD) (chú ý rằng `trainer` đã biết được các tham số nào để tối ưu hóa hơn, nên chúng ta chỉ cần chuyền trong kích thước của minibatch.
```
Comment (lkhphuc) [inline chapter_linear-networks/linear-regression-gluon_vn.md]:
```suggestion
Cập nhật các tham số của mô hình bằng cách gọi bộ tối ưu hoá SGD (chú ý rằng `trainer` đã biết được các tham số nào cần tối ưu, nên chúng ta chỉ cần truyền vô thêm kích thước của minibatch).
```

File: chapter_linear-networks/linear-regression-gluon_vn.md (hunk 4 above)
Comment (tanduong) [reply chapter_linear-networks/linear-regression-gluon_vn.md]: Khi mình viết tiếng việt thì `Tính các gradient` nghe hơi ... lạ. Tiếng việt ít khi phân biệt số ít, số nhiều như tiếng anh và khi không cần thiết, mình nghĩ nên lược bỏ các bổ ngữ chỉ số lượng.

File: chapter_linear-networks/linear-regression-gluon_vn.md (hunk 5 above)
Comment (tanduong) [reply chapter_linear-networks/linear-regression-gluon_vn.md]:
```suggestion
*  Đưa ra dự đoán bằng cách gọi `net(X)` và tính giá trị mất mát `l` (theo chiều thuận).
```

File: chapter_linear-networks/linear-regression-gluon_vn.md (hunk 4 above)
Comment (lkhphuc) [reply chapter_linear-networks/linear-regression-gluon_vn.md]: Mình edit suggestion ở trên theo gợi ý rồi nhé @tanduong

File: chapter_linear-networks/linear-regression-gluon_vn.md [hunk 8]
```diff
@@ -268,29 +268,34 @@ Once we start working with much more complex models, Gluon's advantages will gro
 However, once we have all the basic pieces in place, the training loop itself is strikingly similar to what we did when implementing everything from scratch.
 -->
 
-*dịch đoạn phía trên*
+Bạn có thể thấy rằng việc thể hiện mô hình của chúng ta thông qua Gluon đòi hỏi một vài dòng lệnh tương đối.
+Chúng ta cần phải phân bổ các tham số một cách riêng lẻ, xác định hàm số mất mát của chúng ta , hay hạ gradient ngẫu nhiên.
```
[the remark this answers is not in the document]
Comment (tanduong) [reply chapter_linear-networks/linear-regression-gluon_vn.md]:
```suggestion
Chúng ta không cần khai báo từng tham số một, định nghĩa hàm mất mát, hay tính các đạo hàm riêng.
```

`không cần phải` -> `không cần`
`một cách riêng lẻ` -> `từng cái một`
`implement stochastic gradient descent` là một cụm khó dịch. Nhưng may mắn là cốt lõi của hành động này là tính các đạo hàm riêng. Dịch thế này thì không sát chữ nhưng đúng bản chất.

File: chapter_linear-networks/linear-regression-gluon_vn.md (hunk 1 above)
Comment (tanduong) [reply chapter_linear-networks/linear-regression-gluon_vn.md]: Mạnh dạn lên, viết lại câu luôn:

```suggestion
Lợi ích của Gluon sẽ càng nhiều khi chúng ta làm với những mô hình càng phức tạp.
```

File: chapter_linear-networks/linear-regression-gluon_vn.md (hunk 2 above)
Comment (tanduong) [reply chapter_linear-networks/linear-regression-gluon_vn.md]:
```suggestion
Khi chúng ta có đủ các mảnh ghép quan trọng, vòng huấn luyện sẽ rất giống với vòng huấn luyện chúng ta tự xây dựng mà không dùng Gluon.
```

Comment (tanduong) [reply chapter_linear-networks/linear-regression-gluon_vn.md]: Notes: đừng đùng chữ `triển khai` để dịch chữ `implement`. Dùng chữ `xây dựng` á.

File: chapter_linear-networks/linear-regression-gluon_vn.md [hunk 9]
```diff
@@ -268,29 +268,34 @@ Once we start working with much more complex models, Gluon's advantages will gro
...
+Lợi ích Gluon mang lại càng nhiều khi chúng ta làm với những mô hình càng phức tạp.
+Tuy nhiên, một khi ta nắm được rõ các điều cơ bản, vòng huấn luyện tự thân nó sẽ trông rất giống với những gì chúng ta tự thực hiện từ đầu.
 
 <!--
 To refresh your memory: for some number of epochs, we will make a complete pass over the dataset (train_data), iteratively grabbing one minibatch of inputs and the corresponding ground-truth labels.
 For each minibatch, we go through the following ritual:
 -->
 
-*dịch đoạn phía trên*
+Để nhắc lại: cho một vài epoch, chúng ta sẽ duyêt qua toàn bộ tập dữ liệu (train_data) bằng cách lấy theo từng minibatch của dữ liệu đầu vào cộng với các nhãn gốc tương ứng.
```
Comment (lkhphuc) [inline chapter_linear-networks/linear-regression-gluon_vn.md]:
```suggestion
Để nhắc lại: cho một vài epoch, chúng ta sẽ duyệt qua toàn bộ tập dữ liệu (train_data) bằng cách lấy theo từng minibatch của dữ liệu đầu vào cộng với các nhãn gốc tương ứng.
```

File: chapter_linear-networks/linear-regression-gluon_vn.md [hunk 10]
```diff
@@ -268,29 +268,34 @@ Once we start working with much more complex models, Gluon's advantages will gro
...
 
 <!--
 * Generate predictions by calling `net(X)` and calculate the loss `l` (the forward pass).
 * Calculate gradients by calling `l.backward()` (the backward pass).
 * Update the model parameters by invoking our SGD optimizer (note that `trainer` already knows which parameters to optimize over, so we just need to pass in the minibatch size.
 -->
 
-*dịch đoạn phía trên*
+*  Đưa ra dự đoán bằng cách gọi `net(X)` và tính giá trị mất mát `l` (theo chiều thuận).
+* Tính gradient bằng cách gọi `l.backward()` (đường chuyền ngược).
```
Comment (lkhphuc) [inline chapter_linear-networks/linear-regression-gluon_vn.md]:
```suggestion
* Tính gradient bằng cách gọi `l.backward()` (theo chiều ngược).
```
Chỗ này sửa lại cho thống nhất với forward pass ở bên trên.

File: chapter_linear-networks/linear-regression-gluon_vn.md [hunk 11]
```diff
@@ -268,29 +268,34 @@ Once we start working with much more complex models, Gluon's advantages will gro
...
+Cập nhật các tham số của mô hình bằng cách gọi bộ tối ưu hoá SGD (chú ý rằng `trainer` đã biết được các tham số nào cần tối ưu, nên chúng ta chỉ cần truyền vô thêm kích thước của minibatch).
```
Comment (lkhphuc) [inline chapter_linear-networks/linear-regression-gluon_vn.md]:
```suggestion
* Cập nhật các tham số của mô hình bằng cách gọi bộ tối ưu hoá SGD (chú ý rằng `trainer` đã biết được các tham số nào cần tối ưu, nên chúng ta chỉ cần truyền vô thêm kích thước của minibatch).
```
